code/game/objects/obj_defense.dm

```diff
@@ -27,7 +27,7 @@
 			return 0
 	var/armor_protection = 0
 	if(damage_flag)
-		armor_protection = armor[damage_flag]
+		armor_protection = armor.vars[damage_flag]
```

Comment:
this could be dangerous if an admin changes the `armor` var to a completly different datum and then they would be able to change any var on any datum they would like with no restrictions

Reply:
why is that dangerous, we can't protect very well against admins with var edit access

Reply:
wait, no, for some reason i was reading/thinking about `armor.vars[dmg_flag] = value`, and not just grabbing the value, this is fine and the worst that can happen is a runtime value I guess.

Reply:
This is still bad, some minor vv-fu would allow reading protected configs, which includes the db info.

Reply:
If you're gonna do this, you'd need to verify it actually is an armour datum first